pkg/autoscaler/stats_scraper.go

```diff
@@ -132,22 +143,31 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 
 	sampleSize := populationMeanSampleSize(readyPodsCount)
 	statCh := make(chan *Stat, sampleSize)
+	scrapedPods := &safeStringSet{set: sets.NewString()}
 
 	grp := errgroup.Group{}
 	for i := 0; i < sampleSize; i++ {
 		grp.Go(func() error {
-			stat, err := s.sClient.Scrape(s.url)
-			if err != nil {
-				return err
+			for tries := 0; ; tries++ {
+				stat, err := s.tryScrape(scrapedPods.addIfAbsent)
+				if err != nil {
+					// Return the error if we exhausted our retries.
+					if tries == scraperMaxRetries-1 {
+						return err
+					}
+					time.Sleep(scraperRetryInterval)
+					continue
```

Comment:
Perhaps we should differentiate error: having a duplicate pod, does not call for a sleep, I'd just fire next one immediately. Or, maybe, we don't need sleep at all, hoping we'll hit a different pod?

Reply:
I'd say let's remove the sleep altogether.